--- mesonbuild/compilers/mixins/clike.py.orig	2024-07-26 20:05:04.000000000 +0200
+++ mesonbuild/compilers/mixins/clike.py	2024-09-05 10:40:15.000000000 +0200
@@ -1184,14 +1184,13 @@
         unless you select a particular macOS SDK with the -isysroot flag.
         You can also add to this by setting -F in CFLAGS.
         '''
-        # TODO: this really needs to be *AppleClang*, not just any clang.
-        if self.id != 'clang':
-            raise mesonlib.MesonException('Cannot find framework path with non-clang compiler')
+        # TODO: should probably check for macOS?
         # Construct the compiler command-line
         commands = self.get_exelist(ccache=False) + ['-v', '-E', '-']
         commands += self.get_always_args()
         # Add CFLAGS/CXXFLAGS/OBJCFLAGS/OBJCXXFLAGS from the env
-        commands += env.coredata.get_external_args(self.for_machine, self.language)
+        # 'std=c++<ver>','std=gnu++<ver>' are valid for C++/ObjC++ but not for 'C'
+        commands += [x for x in env.coredata.get_external_args(self.for_machine, self.language) if '-std=c++' not in x and '-std=gnu++' not in x]
         mlog.debug('Finding framework path by running: ', ' '.join(commands), '\n')
         os_env = os.environ.copy()
         os_env['LC_ALL'] = 'C'
