--- src/libtracker-miners-common/tracker-term-utils.c.org	2021-04-02 23:32:31.000000000 +0200
+++ src/libtracker-miners-common/tracker-term-utils.c	2024-04-29 18:25:03.000000000 +0200
@@ -138,8 +138,13 @@
 	if (!tracker_term_is_tty ())
 		return FALSE;
 
-	if (pipe2 (fds, O_CLOEXEC) < 0)
-		return FALSE;
+#ifdef __APPLE__
+    if (pipe(fds) < 0 )
+        return FALSE;
+#else
+ 	if (pipe2 (fds, O_CLOEXEC) < 0)
+ 		return FALSE;
+#endif
 
 	pager_command = best_pager ();
 	if (!pager_command)
