--- gedit/gedit-app-osx.m.orig	2021-03-27 18:41:09.000000000 +0100
+++ gedit/gedit-app-osx.m	2024-10-01 10:24:20.000000000 +0200
@@ -225,12 +225,12 @@
 
 		if (link_id)
 		{
-			link = g_strdup_printf ("http://library.gnome.org/users/gedit/stable/%s",
+			link = g_strdup_printf ("https://gedit-technology.github.io/user-manuals/gedit/%s",
 						link_id);
 		}
 		else
 		{
-			link = g_strdup ("http://library.gnome.org/users/gedit/stable/");
+			link = g_strdup ("https://gedit-technology.github.io/user-manuals/gedit/");
 		}
 
 		ret = gedit_app_osx_show_url (GEDIT_APP_OSX (app), link);
@@ -506,7 +506,7 @@
 
 	window = GEDIT_APP_CLASS (gedit_app_osx_parent_class)->create_window (app);
 
-	gtk_window_set_titlebar (GTK_WINDOW (window), NULL);
+/*	gtk_window_set_titlebar (GTK_WINDOW (window), NULL); */
 
 	if (gtk_widget_get_realized (GTK_WIDGET (window)))
 	{
@@ -595,7 +595,7 @@
 	   command line arguments by itself and generate OpenFile events.
 	   We already parse the command line ourselves, so this is needed
 	   to prevent opening files twice, etc. */
-	[[NSUserDefaults standardUserDefaults] setObject:@"NO"
+	[[NSUserDefaults standardUserDefaults] setObject:@"YES"
 	                                       forKey:@"NSTreatUnknownArgumentsAsOpen"];
 }
 
