--- gedit/gedit-app-osx.m.orig	2021-03-27 18:41:09.000000000 +0100
+++ gedit/gedit-app-osx.m	2024-09-30 04:13:30.000000000 +0200
@@ -31,7 +31,8 @@
 #include "gedit-debug.h"
 #include "gedit-commands.h"
 #include "gedit-commands-private.h"
-#include "gedit-recent.h"
+#include "gedit-document-private.h"
+#include "gedit-recent-osx.h"
 #import <AppKit/AppKit.h>
 
 NSWindow *gdk_quartz_window_get_nswindow(GdkWindow *window);
@@ -225,12 +226,12 @@
 
 		if (link_id)
 		{
-			link = g_strdup_printf ("http://library.gnome.org/users/gedit/stable/%s",
+			link = g_strdup_printf ("https://gedit-technology.github.io/user-manuals/gedit/%s",
 						link_id);
 		}
 		else
 		{
-			link = g_strdup ("http://library.gnome.org/users/gedit/stable/");
+			link = g_strdup ("https://gedit-technology.github.io/user-manuals/gedit/");
 		}
 
 		ret = gedit_app_osx_show_url (GEDIT_APP_OSX (app), link);
@@ -287,6 +288,7 @@
 		}
 
 		ismodified = !gedit_document_is_untouched (document);
+/*		ismodified = !tepl_buffer_is_untouched (TEPL_BUFFER (document)); */
 		[native setDocumentEdited:ismodified];
 	}
 	else
@@ -506,7 +508,7 @@
 
 	window = GEDIT_APP_CLASS (gedit_app_osx_parent_class)->create_window (app);
 
-	gtk_window_set_titlebar (GTK_WINDOW (window), NULL);
+/*	gtk_window_set_titlebar (GTK_WINDOW (window), NULL); */
 
 	if (gtk_widget_get_realized (GTK_WIDGET (window)))
 	{
