--- gedit/resources/gtk/menus-common.ui.orig	2021-03-27 18:41:09.000000000 +0100
+++ gedit/resources/gtk/menus-common.ui	2024-10-01 08:02:09.000000000 +0200
@@ -166,7 +166,6 @@
           </item>
           <item>
             <attribute name="label" translatable="yes">_Quit</attribute>
-            <attribute name="hidden-when">macos-menubar</attribute>
             <attribute name="action">app.quit</attribute>
             <attribute name="accel">&lt;Primary&gt;Q</attribute>
           </item>
@@ -228,7 +227,6 @@
         <section>
           <item>
             <attribute name="label" translatable="yes">_Preferences</attribute>
-            <attribute name="hidden-when">macos-menubar</attribute>
             <attribute name="action">app.preferences</attribute>
           </item>
         </section>
@@ -392,7 +390,6 @@
         <section>
           <item>
             <attribute name="label" translatable="yes">_About Text Editor</attribute>
-            <attribute name="hidden-when">macos-menubar</attribute>
             <attribute name="action">app.about</attribute>
           </item>
         </section>
--- plugins/externaltools/tools/appactivatable.py.orig	2021-03-27 18:41:10.000000000 +0100
+++ plugins/externaltools/tools/appactivatable.py	2024-10-01 08:31:47.000000000 +0200
@@ -62,7 +62,7 @@
             # FIXME: find a better way to share the action name
             action_name = 'external-tool-%X-%X' % (id(tool), id(tool.name))
             item = Gio.MenuItem.new(tool.name.replace('_', '__'), "win.%s" % action_name)
-            item.set_attribute_value("hidden-when", GLib.Variant.new_string("action-disabled"))
+            #item.set_attribute_value("hidden-when", GLib.Variant.new_string("action-disabled"))
             menu.append_item(item)
 
             if tool.shortcut:
