--- meson.build.orig	2021-03-27 18:41:10.000000000 +0100
+++ meson.build	2024-09-21 09:50:53.000000000 +0200
@@ -1,7 +1,7 @@
 project(
   'gedit', 'c',
   version: '3.38.2',
-  meson_version: '>= 0.53'
+  meson_version: '>= 0.59'
 )
 
 if host_machine.system() == 'darwin'
@@ -26,7 +26,7 @@
 
 # Subprojects
 if not get_option('buildtype').contains('plain')
-  run_command('git', '-C', meson.source_root(), 'submodule', 'update', '--init', '--recursive')
+  run_command('git', '-C', meson.project_source_root(), 'submodule', 'update', '--init', '--recursive')
 endif
 
 libgd_subproject = subproject(
@@ -55,7 +55,7 @@
 ]
 
 gspell_dep = dependency('gspell-1', version: '>= 1.0')
-python3 = python.find_installation('python3')
+python3 = python.find_installation('@@PYTHON3@@')
 
 # Configurations
 config_h = configuration_data()
