--- gsk/gpu/shaders/meson.build.orig	2024-04-19 03:30:02.000000000 +0200
+++ gsk/gpu/shaders/meson.build	2024-10-04 06:30:20.000000000 +0200
@@ -62,7 +62,7 @@
                              capture: true)
   gsk_private_gpu_gl_shaders += gl_shader
 
-  if (have_vulkan)
+  if (have_vulkan) and (have_glslc)
     glslc_options = [
       [ fs.name (fs.replace_suffix (shader, '')) + '.1.2.vert.spv',
         [ '--target-env=vulkan1.2',
