--- meson.build.orig	2024-04-19 03:30:02.000000000 +0200
+++ meson.build	2024-10-04 06:29:10.000000000 +0200
@@ -330,7 +330,7 @@
     'write-strings',
   ]
 
-  if get_option('buildtype').startswith('debug')
+  if get_option('buildtype') == 'debug'
     foreach warning: extra_warnings
       test_cflags += '-Werror=@0@'.format(warning)
     endforeach
@@ -628,6 +628,12 @@
   vulkan_pkg_found = false
 endif
 
+if glslc.found()
+    have_glslc = true
+else
+    have_glslc = false
+endif
+
 libdrm_dep = dependency('libdrm', required: os_linux)
 # We only care about drm_fourcc.h for all the fourccs,
 # but not about linking to libdrm
@@ -869,6 +875,18 @@
 )
 meson.override_dependency('gtk4', libgtk_dep)
 
+if not x11_enabled
+  pkg = 'gtk4-quartz'
+  pkg_config.generate(
+    filebase: pkg,
+    unescaped_variables: common_pc_variables,
+    name: 'GTK',
+    description: 'GTK Graphical UI Library',
+    requires: ['gtk4', get_variable('@0@_public_deps'.format(backend), [])],
+  )
+  meson.override_dependency(pkg, libgtk_dep)
+endif
+
 foreach backend: enabled_backends
   pkg = 'gtk4-@0@'.format(backend)
   pkg_config.generate(
