--- src/libtracker-common/tracker-term-utils.c.org	2020-12-09 00:01:22.000000000 +0100
+++ src/libtracker-common/tracker-term-utils.c	2024-04-28 13:31:56.000000000 +0200
@@ -138,8 +138,13 @@
 	if (!tracker_term_is_tty ())
 		return FALSE;
 
+#ifdef __APPLE__
+    if (pipe(fds) < 0 )
+        return FALSE;
+#else
 	if (pipe2 (fds, O_CLOEXEC) < 0)
 		return FALSE;
+#endif
 
 	pager_command = best_pager ();
 	if (!pager_command)
