--- sys/applemedia/vtdec.c.orig	2024-08-21 13:25:15.000000000 +0200
+++ sys/applemedia/vtdec.c	2024-08-29 10:18:51.000000000 +0200
@@ -1216,12 +1216,12 @@
 
   if (status != noErr) {
     switch (status) {
-      case kVTVideoDecoderReferenceMissingErr:
+      /*case kVTVideoDecoderReferenceMissingErr: cvr removed not existing anymore.
         /* ReferenceMissingErr is not critical, when it occurs the frame
-         * usually has the kVTDecodeInfo_FrameDropped flag set. Log only for debugging purposes. */
+         * usually has the kVTDecodeInfo_FrameDropped flag set. Log only for debugging purposes. 
         GST_DEBUG_OBJECT (vtdec, "ReferenceMissingErr when decoding frame %d",
             frame->decode_frame_number);
-        break;
+        break; */
 #ifndef HAVE_IOS
       case codecBadDataErr:    /* SW decoder on macOS uses a different code from the hardware one... */
 #endif
