--- ext/cdparanoia/meson.build.org	2024-03-21 21:47:53.000000000 +0100
+++ ext/cdparanoia/meson.build	2024-04-14 13:29:05.000000000 +0200
@@ -11,10 +11,10 @@
   cdparanoia_deps = [cdparanoia_dep]
   cdparanoia_found = true
 else
-  cdparanoia_dep = cc.find_library('cdda_paranoia', required : cdparanoia_option)
-  cdinterface_dep = cc.find_library('cdda_interface', required : cdparanoia_option)
-  if cdparanoia_dep.found() and cdinterface_dep.found()
-    cdparanoia_deps = [cdparanoia_dep, cdinterface_dep]
+  cdparanoia_lib = cc.find_library('cdda_paranoia', required : cdparanoia_option)
+  cdinterface_lib = cc.find_library('cdda_interface', required : cdparanoia_option)
+  if cdparanoia_lib.found() and cdinterface_lib.found()
+    cdparanoia_deps = declare_dependency(  link_args : ['-Wl,-rpath,/opt/local/lib /opt/local/lib/libcdda_paranoia.dylib /opt/local/lib/libcdda_interface.dylib'] )
     cdparanoia_found = true
     if cc.has_header_symbol('cdda/cdda_interface.h', 'cdda_open')
       core_conf.set('CDPARANOIA_HEADERS_IN_DIR', true)
@@ -43,7 +43,7 @@
     ['gstcdparanoiasrc.c'],
     include_directories: [configinc, libsinc],
     c_args : gst_plugins_base_args + extra_args,
-    dependencies : cdparanoia_deps + [audio_dep, gst_dep, gst_base_dep],
+    dependencies : [cdparanoia_deps] + [audio_dep,gst_dep,gst_base_dep],
     install : true,
     install_dir : plugins_install_dir,
   )
