--- meson.build.orig	2024-08-15 13:35:14.000000000 +0200
+++ meson.build	2024-08-16 08:10:53.000000000 +0200
@@ -111,8 +111,9 @@
   gio_dep,
   dependency('gtk+-3.0', version: '>= 3.22'),
   dependency('libgedit-amtk-5'),
-  dependency('libgedit-tepl-6'),
+  dependency('tepl-6'),
   dependency('libpeas-gtk-1.0'),
+  dependency('libgedit-gfls-1'),
 ]
 
 deps_basic_list = [
--- docs/reference/meson.build.org	2024-08-15 13:50:37.000000000 +0200
+++ docs/reference/meson.build	2024-08-16 06:01:47.000000000 +0200
@@ -7,7 +7,7 @@
 gtk_docpath = dependency('gtk+-3.0').get_variable(pkgconfig: 'prefix') / 'share/gtk-doc/html/gtk3'
 gsv_docpath = dependency('libgedit-gtksourceview-300').get_variable(pkgconfig: 'prefix') / 'share/gtk-doc/html/libgedit-gtksourceview-300'
 amtk_docpath = dependency('libgedit-amtk-5').get_variable(pkgconfig: 'prefix') / 'share/gtk-doc/html/libgedit-amtk-5'
-tepl_docpath = dependency('libgedit-tepl-6').get_variable(pkgconfig: 'prefix') / 'share/gtk-doc/html/libgedit-tepl-6'
+tepl_docpath = dependency('tepl-6').get_variable(pkgconfig: 'prefix') / 'share/gtk-doc/html/tepl-6'
 libpeas_docpath = dependency('libpeas-1.0').get_variable(pkgconfig: 'prefix') / 'share/gtk-doc/html/libpeas'
 
 gedit_doc_dep = declare_dependency(
--- gedit/meson.build.orig	2024-08-15 13:34:46.000000000 +0200
+++ gedit/meson.build	2024-08-16 08:18:38.000000000 +0200
@@ -189,7 +189,7 @@
   symbol_prefix: 'gedit',
   identifier_prefix: 'Gedit',
   export_packages: 'gedit-@0@'.format(api_version),
-  includes: ['Gtk-3.0', 'GtkSource-300', 'Tepl-6'],
+  includes: ['Gtk-3.0', 'GtkSource-300', 'Tepl-6', 'Gfls-1'],
   link_with: libgedit_shared_lib,
   install: true,
   install_dir_gir: get_option('datadir') / 'gedit/gir-1.0',
