--- meson.build.orig	2024-09-14 07:18:34.000000000 +0200
+++ meson.build	2024-11-07 09:53:51.000000000 +0100
@@ -2,11 +2,10 @@
   'gedit', 'c',
   version: '48.0',
   meson_version: '>= 0.59',
-  default_options: ['warning_level=2']
 )
 
 if host_machine.system() == 'darwin'
-  add_languages('objc')
+  add_languages('objc', native: false)
 endif
 
 gnome = import('gnome')
@@ -115,7 +114,7 @@
 ]
 
 gspell_dep = dependency('gspell-1', version: '>= 1.0')
-python3 = python.find_installation('python3')
+python3 = python.find_installation('/opt/local/bin/python3.12')
 
 # Configurations
 config_h = configuration_data()
@@ -153,6 +152,7 @@
 meson.add_install_script(
   'build-aux/meson/post_install.py',
   get_option('prefix') / get_option('libdir'),
+  get_option('prefix') / get_option('datadir'),
 )
 
 gnome.post_install(
