--- gedit/gedit-app.c.orig	2024-04-27 17:29:08.000000000 +0200
+++ gedit/gedit-app.c	2024-08-16 07:36:13.000000000 +0200
@@ -669,7 +669,7 @@
 					     GEDIT_SETTINGS_USE_DEFAULT_FONT,
 					     GEDIT_SETTINGS_EDITOR_FONT);
 
-	tepl_settings_handle_simple_theme_variant (tepl_settings,
+	tepl_settings_handle_theme_variant (tepl_settings,
 						   ui_settings,
 						   GEDIT_SETTINGS_THEME_VARIANT);
 }
