--- src/cocoa_menu_item.c.orig	2024-11-16 15:06:44.000000000 +0100
+++ src/cocoa_menu_item.c	2024-11-19 09:50:36.000000000 +0100
@@ -21,6 +21,7 @@
  */
 
 #include <config.h>
+#include <gdk/gdkquartz.h> //for gdk_quartz_osx_version()
 #import <Cocoa/Cocoa.h>
 #include <gtk/gtk.h>
 #if GTK_MAJOR_VERSION == 3
@@ -39,6 +40,33 @@
 //#define DEBUG(format, ...) g_printerr ("%s: " format, G_STRFUNC, ## __VA_ARGS__)
 #define DEBUG(format, ...)
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 101200
+#define GDK_QUARTZ_ALTERNATE_KEY_MASK NSAlternateKeyMask
+#define GDK_QUARTZ_COMMAND_KEY_MASK NSCommandKeyMask
+#define GDK_QUARTZ_CONTROL_KEY_MASK NSControlKeyMask
+#define GDK_QUARTZ_SHIFT_KEY_MASK NSShiftKeyMask
+#define GDK_QUARTZ_KEY_DOWN NSKeyDown
+#else
+#define GDK_QUARTZ_ALTERNATE_KEY_MASK NSEventModifierFlagOption
+#define GDK_QUARTZ_COMMAND_KEY_MASK NSEventModifierFlagCommand
+#define GDK_QUARTZ_CONTROL_KEY_MASK NSEventModifierFlagControl
+#define GDK_QUARTZ_SHIFT_KEY_MASK NSEventModifierFlagShift
+#define GDK_QUARTZ_KEY_DOWN NSEventTypeKeyDown
+#endif
+
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 101300
+#define GDK_QUARTZ_MIXED_STATE NSMixedState
+#define GDK_QUARTZ_ON_STATE NSOnState
+#define GDK_QUARTZ_OFF_STATE NSOffState
+#define GDK_QUARTZ_NUMERIC_PAD_KEY_MASK NSNumericPadKeyMask
+#else
+#define GDK_QUARTZ_MIXED_STATE NSControlStateValueMixed
+#define GDK_QUARTZ_ON_STATE NSControlStateValueOn
+#define GDK_QUARTZ_OFF_STATE NSControlStateValueOff
+#define GDK_QUARTZ_NUMERIC_PAD_KEY_MASK NSEventModifierFlagNumericPad
+#endif
+
+
 /*
  * These functions are long, ugly, and monotonous, so forward declare
  * them here and define them at the end of the file.
@@ -123,11 +151,11 @@
                 NULL);
 
   if (inconsistent)
-    [cocoa_item setState: NSMixedState];
+    [cocoa_item setState: GDK_QUARTZ_MIXED_STATE];
   else if (active)
-    [cocoa_item setState: NSOnState];
+    [cocoa_item setState: GDK_QUARTZ_ON_STATE];
   else
-    [cocoa_item setState: NSOffState];
+    [cocoa_item setState: GDK_QUARTZ_OFF_STATE];
 }
 
 static void
@@ -249,7 +277,7 @@
 		  [cocoa_item setKeyEquivalent: @""];
                   return;
                 }
-              modifiers |= NSNumericPadKeyMask;
+              modifiers |= GDK_QUARTZ_NUMERIC_PAD_KEY_MASK;
             }
 
           /* if we somehow got here with GDK_A ... GDK_Z rather than GDK_a ... GDK_z, then take note
@@ -258,7 +286,7 @@
 
           if (keyval_is_uppercase (actual_key))
             {
-              modifiers |= NSShiftKeyMask;
+              modifiers |= GDK_QUARTZ_SHIFT_KEY_MASK;
             }
 
           str = gdk_quartz_keyval_to_string (actual_key);
@@ -286,24 +314,24 @@
             {
               if (key->accel_mods & GDK_SHIFT_MASK)
                 {
-                  modifiers |= NSShiftKeyMask;
+                  modifiers |= GDK_QUARTZ_SHIFT_KEY_MASK;
                 }
 
               if (key->accel_mods & GDK_CONTROL_MASK)
                 {
-                  modifiers |= NSControlKeyMask;
+                  modifiers |= GDK_QUARTZ_CONTROL_KEY_MASK;
                 }
 
               /* gdk/quartz maps Alt/Option to Mod5 */
               if (key->accel_mods & (GDK_MOD1_MASK))
                 {
-                  modifiers |= NSAlternateKeyMask;
+                  modifiers |= GDK_QUARTZ_ALTERNATE_KEY_MASK;
                 }
 
               /* gdk/quartz maps Command to MOD1 */
               if (key->accel_mods & GDK_META_MASK)
                 {
-                  modifiers |= NSCommandKeyMask;
+                  modifiers |= GDK_QUARTZ_COMMAND_KEY_MASK;
                 }
 
             }
