--- src/gtkosxapplication_quartz.c.orig	2024-11-20 08:21:30.000000000 +0100
+++ src/gtkosxapplication_quartz.c	2024-11-20 08:39:07.000000000 +0100
@@ -68,7 +68,9 @@
  * - Figure out what to do per app/window...
  *
  */
-G_DEFINE_TYPE (GtkosxApplication, gtkosx_application, G_TYPE_OBJECT)
+//G_DEFINE_TYPE (GtkosxApplication, gtkosx_application, G_TYPE_OBJECT)
+G_DEFINE_TYPE_WITH_CODE (GtkosxApplication, gtkosx_application, G_TYPE_OBJECT,
+    G_ADD_PRIVATE(GtkosxApplication));
 
 static GQuark emission_hook_quark = 0;
 
@@ -595,8 +597,10 @@
 static void
 gtkosx_application_init (GtkosxApplication *self)
 {
+  g_message ("CVR XDEBUG gtkosx_application_init (GtkosxApplication *self) is used");
   [NSApplication sharedApplication];
-  self->priv = GTKOSX_APPLICATION_GET_PRIVATE (self);
+  //self->priv = GTKOSX_APPLICATION_GET_PRIVATE (self);
+  self->priv = (GtkosxApplicationPrivate *) gtkosx_application_get_instance_private (self);
   self->priv->use_quartz_accelerators = TRUE;
   self->priv->dock_menu = NULL;
   gdk_window_add_filter (NULL, global_event_filter_func, (gpointer)self);
@@ -644,7 +648,7 @@
 gtkosx_application_class_init (GtkosxApplicationClass *klass)
 {
   GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
-  g_type_class_add_private (klass, sizeof (GtkosxApplicationPrivate));
+  //g_type_class_add_private (klass, sizeof (GtkosxApplicationPrivate));
   gobject_class->constructor = gtkosx_application_constructor;
   /**
    * GtkosxApplication::NSApplicationDidBecomeActive:
--- src/gtkosxapplicationprivate.h.orig	2024-11-20 08:21:30.000000000 +0100
+++ src/gtkosxapplicationprivate.h	2024-11-20 08:29:15.000000000 +0100
@@ -29,7 +29,7 @@
 #import "GtkApplicationNotify.h"
 #import "GtkApplicationDelegate.h"
 
-#define  GTKOSX_APPLICATION_GET_PRIVATE(obj)	(G_TYPE_INSTANCE_GET_PRIVATE ((obj), GTKOSX_TYPE_APPLICATION, GtkosxApplicationPrivate))
+//#define  GTKOSX_APPLICATION_GET_PRIVATE(obj)	(G_TYPE_INSTANCE_GET_PRIVATE ((obj), GTKOSX_TYPE_APPLICATION, GtkosxApplicationPrivate))
 
 struct _GtkosxApplicationPrivate
 {
--- configure.ac.orig	2024-11-20 08:21:30.000000000 +0100
+++ configure.ac	2024-11-20 10:26:33.000000000 +0100
@@ -35,8 +35,7 @@
 AC_SUBST([GETTEXT_PACKAGE], [gtk-mac-integration])
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE], "$GETTEXT_PACKAGE", [Define to the gettext package name.])
 
-AM_PATH_GLIB_2_0([2.14.0])
-AC_MSG_CHECKING([GLib Version])
+AM_PATH_GLIB_2_0([2.38.0])
 PKG_INSTALLDIR()
 
 GTK_DOC_CHECK([1.11], [--flavour no-tmpl])
