diff -ur gst-libav-1.24.1.org/meson.build gst-libav-1.24.1/meson.build
--- gst-libav-1.24.1.org/meson.build	2024-04-30 13:11:33.000000000 +0200
+++ meson.build	2024-04-30 15:07:38.000000000 +0200
@@ -18,22 +18,18 @@
 gst_version_is_dev = gst_version_minor.is_odd() and gst_version_micro < 90
 
 api_version = '1.0'
-libavfilter_dep = dependency('libavfilter', version: '>= 7.16.100',
-  fallback: ['FFmpeg', 'libavfilter_dep'])
-libavformat_dep = dependency('libavformat', version: '>= 58.12.100',
-  fallback: ['FFmpeg', 'libavformat_dep'])
-libavcodec_dep = dependency('libavcodec', version: '>= 58.18.100',
-  fallback: ['FFmpeg', 'libavcodec_dep'])
-libavutil_dep = dependency('libavutil', version: '>= 56.14.100',
-  fallback: ['FFmpeg', 'libavutil_dep'])
+libavfilter_dep = dependency('libavfilter', version: '>= 9.12.100', method : 'pkg-config')
+libavformat_dep = dependency('libavformat', version: '>= 60.16.100', method : 'pkg-config')
+libavcodec_dep = dependency('libavcodec', version: '>= 60.31.102', method : 'pkg-config')
+libavutil_dep = dependency('libavutil', version: '>= 58.29.100', method : 'pkg-config')
 
 libav_deps = [libavfilter_dep, libavformat_dep, libavcodec_dep, libavutil_dep]
 
 cc = meson.get_compiler('c')
 static_build = get_option('default_library') == 'static'
 
-check_ffmpeg_src = '''#include <libavcodec/avcodec.h>
-#if LIBAVCODEC_VERSION_MICRO >= 100
+check_ffmpeg_src = '''#include "/opt/local/libexec/ffmpeg6/include/libavcodec/version.h"
+#if LIBAVCODEC_VERSION_MICRO >= 102
 /* FFmpeg uses 100+ as its micro version */
 #else
 #error libav provider should be FFmpeg
