--- plugins/findinfiles/dialog.vala.orig	2021-03-27 18:53:16.000000000 +0100
+++ plugins/findinfiles/dialog.vala	2024-09-20 09:50:18.000000000 +0200
@@ -22,17 +22,17 @@
 [GtkTemplate (ui = "/org/gnome/gedit/plugins/findinfiles/ui/dialog.ui")]
 class FindDialog : Gtk.Dialog {
     [GtkChild]
-    public Gtk.Entry search_entry;
+    public unowned Gtk.Entry search_entry;
     [GtkChild]
-    public Gtk.FileChooserButton sel_folder;
+    public unowned Gtk.FileChooserButton sel_folder;
     [GtkChild]
-    public Gtk.CheckButton match_case_checkbutton;
+    public unowned Gtk.CheckButton match_case_checkbutton;
     [GtkChild]
-    public Gtk.CheckButton entire_word_checkbutton;
+    public unowned Gtk.CheckButton entire_word_checkbutton;
     [GtkChild]
-    public Gtk.CheckButton regex_checkbutton;
+    public unowned Gtk.CheckButton regex_checkbutton;
     [GtkChild]
-    public Gtk.Widget find_button;
+    public unowned Gtk.Widget find_button;
 
     public FindDialog (File? root) {
         if (root != null) {
