--- src/GNSMenuItem.c.orig	2021-06-29 02:49:49.000000000 +0200
+++ src/GNSMenuItem.c	2024-05-12 15:19:34.000000000 +0200
@@ -44,6 +44,7 @@
 
   /* All menu items have the action "activate", which will be handled by this child class
    */
+  const char* locale = setlocale(LC_NUMERIC, "C");
   self = [ super initWithTitle: title action: @selector (activate: ) keyEquivalent: @"" ];
 
   if (self)
@@ -58,6 +59,7 @@
       notUsed = NO;
       g_weak_ref_set (&menuItem, GTK_MENU_ITEM (gtkmenuitem));
     }
+  setlocale(LC_NUMERIC, locale);
   return self;
 }
 
